stdlib/config-api/src/test/java/org/ballerinalang/stdlib/config/AESCipherToolTest.java

```diff
@@ -60,4 +60,23 @@ public void testEncryptionAndDecryptionWithLongSecret() throws AESCipherToolExce
         String encryptedStr = aesCipherTool.encrypt(plainText);
         Assert.assertEquals(plainText, aesCipherTool.decrypt(encryptedStr));
     }
+
+    @Test
+    public void testEncryptionAndDecryptionWithTwoCipherToolsNegative() {
+        String plainText = "this is the plain text";
+        String encryptionKey = "abc&xyz";
+        String decryptionKey = "xyz&abc";
+        try {
+            AESCipherTool encryptionCipherTool = new AESCipherTool(encryptionKey);
+            String encryptedStr = encryptionCipherTool.encrypt(plainText);
+            AESCipherTool decryptionCipherTool = new AESCipherTool(decryptionKey);
+            String decryptedText = decryptionCipherTool.decrypt(encryptedStr);
+            // We expect the above line should throw an exception since the decryption key is invalid. But in some cases
+            // the given decryption key is able to decrypt the given encrypted string with the use of padding. Then, the
+            // output (decrypted text) does not match with the plain text.
+            Assert.assertNotEquals(decryptedText, plainText);
+        } catch (AESCipherToolException e) {
```

Comment:
Can't we make this an expected exception?

Reply:
Since it does not throw an exception always, we can't do that. The reason for the intermittent failure was also not throwing exception always.